prinz/src/main/scala/pl/touk/nussknacker/prinz/mlflow/repository/MLFRepositoryRestClient.scala

```diff
@@ -0,0 +1,41 @@
+package pl.touk.nussknacker.prinz.mlflow.repository
+
+import java.net.URL
+
+import pl.touk.nussknacker.prinz.mlflow.model.RegisteredModel
+import pl.touk.nussknacker.prinz.mlflow.MlflowConstants.{BASE_API_PATH, BASE_PREVIEW_API_PATH}
+import pl.touk.nussknacker.prinz.model.repository.ModelRepository
+import pl.touk.nussknacker.prinz.util.http.{RestClientException, RestJsonClient}
+
+
+class MLFRepositoryRestClient(hostUrl: URL) extends ModelRepository {
+  type RestResponse[RESPONSE] = Either[RestClientException, RESPONSE]
+
+  private val restClient = new RestJsonClient(s"$hostUrl$BASE_API_PATH")
+
+  private val previewRestClient = new RestJsonClient(s"$hostUrl$BASE_PREVIEW_API_PATH")
+
+  override def listModels(): RestResponse[List[RegisteredModel]] =
+    previewRestClient.getJson[GetAllRegisteredModelsResponse]("/registered-models/list").right.map(_.registered_models)
+
+  def listModels(maxResults: Int, page: Int): RestResponse[List[RegisteredModel]] =
+    previewRestClient.getJsonBody[GetAllRegisteredModelsResponse]("/registered-models/list",
+      listModelsBody(maxResults, page)).right.map(_.registered_models)
+
+  override def getModel(name: String): RestResponse[RegisteredModel] =
```

Comment:
case class MlfowModelName can be used with some implicit converters, not sure here if string is the most proper. Let's see maybe to some another projects